AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -0,0 +1,886 @@
+//
+//  ExampleAVAudioEngineDevice.m
+//  AudioDeviceExample
+//
+//  Copyright © 2018 Twilio, Inc. All rights reserved.
+//
+
+#import "ExampleAVAudioEngineDevice.h"
+
+// We want to get as close to 10 msec buffers as possible because this is what the media engine prefers.
+static double const kPreferredIOBufferDuration = 0.01;
+
+// We will use mono playback and recording where available.
+static size_t const kPreferredNumberOfChannels = 1;
+
+// An audio sample is a signed 16-bit integer.
+static size_t const kAudioSampleSize = 2;
+static uint32_t const kPreferredSampleRate = 48000;
+
+/*
+ * Calls to AudioUnitInitialize() can fail if called back-to-back after format change or add remove tracks
+ * audio deivce instances. A fall-back solution is to allow multiple sequential calls
+ * with as small delay between each. This factor sets the max number of allowed
+ * initialization attempts.
```

Comment:
Calls to AudioUnitInitialize() can fail if called back-to-back after a format change or adding and removing tracks.

as small -> a small